### PR TITLE
feat: PortfolioView: Add feature flag check for polling intervals

### PR DESCRIPTION
**Pull request by darkwing**
## **Description**

Adds a remote endpoint check to see if the remote API wants us to update the number of seconds between polling intervals for balances.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28501?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Open the extension
2. See the network request being made in the Networks devtools panel
3. Change conditional in response to `if (true) {` and hardcode a `pollInterval` value
4. See `this.tokenBalancesController.setIntervalLength` successfully called.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -2571,6 +2572,38 @@ export default class MetamaskController extends EventEmitter {
...
+    fetchWithCache({
+      url: 'https://swap.api.cx.metamask.io/featureFlags',
+      cacheRefreshTime: MINUTE * 20,
+    })
+      .then((response) => {
+        console.log('getInfuraFeatureFlags response:', response);
+        const { pollInterval, pausedChainIds } = response;
+        // Polling interval is provided in seconds
+        if (pollInterval > 0) {
+          this.tokenBalancesController.setIntervalLength(pollInterval * SECOND);
```
Comment (bergeron) [inline app/scripts/metamask-controller.js:2593]: Will require https://github.com/MetaMask/metamask-extension/pull/28472 before `this.tokenBalancesController` is defined

Comment (darkwing) [reply app/scripts/metamask-controller.js:2619]: PR was merged and I've rebased this PR.

**Comment by darkwing** [comment]
Feature flag implementation here: https://github.com/consensys-vertical-apps/va-mmcx-swap-api/pull/221

**Comment by gambinish** [comment]
We still planning to include this in the release? Or are we deciding to table this in favor of LaunchDarkly?

**Comment by darkwing** [comment]
This needs to go into release.

**Review by gambinish — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Pull request merged**